Packs/FeedProofPoint/Integrations/FeedProofPoint/FeedProofPoint.py

```diff
@@ -0,0 +1,166 @@
+            indicator_type, self.indicator_types_to_endpoint.values()
+        )
+        for endpoint in endpoints:
+            resp = self._http_request(
+                "GET", endpoint, resp_type="text", timeout=(30, 60)
+            )
+            resp = resp.splitlines()
+            csv_repr = csv.reader(resp)
+            headers: list = next(csv_repr, None)
+            headers = [header.replace(" ", "").replace("(|)", "") for header in headers]
+            for line in csv_repr:
+                yield {headers[i]: line[i] for i in range(len(headers))}
+
+    def _build_iterator_domain(self) -> Generator[dict, None, None]:
+        """Gives the user an
+
+        Returns:
+
+        """
+        return self._build_iterator(self.DOMAIN_TYPE)
+
+    def _build_iterator_ip(self) -> Generator[dict, None, None]:
+        return self._build_iterator("ip")
+
+    def get_indicators_domain(self) -> List[dict]:
+        return [
+            {
+                "value": item["domain"],
+                "type": FeedIndicatorType.Domain,
+                "port": item["ports"],
+                "rawJSON": item,
+            }
+            for item in self._build_iterator_domain()
+        ]
+
+    def get_indicators_ip(self) -> List[dict]:
+        return [
+            {
+                "value": item["ip"],
+                "type": FeedIndicatorType.IP,
+                "port": item["ports"],
+                "rawJSON": item,
+            }
+            for item in self._build_iterator_ip()
+        ]
+
+    def get_indicators(self) -> List[dict]:
+        return self.get_indicators_domain() + self.get_indicators_ip()
+
+
+def url_concat(*args) -> str:
+    if args:
+        url = "/".join(element.strip("/") for element in args if element)
+        return url + "/" if args[-1].endswith("/") else url
+    return ""
+
+
+def module_test_command(client: Client, indicator_type: str) -> str:
+    fetch_indicators_command(client, indicator_type)
+    return "ok"
+
+
+def fetch_indicators_command(client: Client, indicator_type: Optional[str]):
+    if indicator_type == client.IP_TYPE:
+        return client.get_indicators_ip()
+    elif indicator_type == client.DOMAIN_TYPE:
+        return client.get_indicators_domain()
+    else:
+        return client.get_indicators()
+
+
+def get_indicators_command(client, args) -> Tuple[str, dict, list]:
+    indicator_type = args.get("indicator_type")
+    if indicator_type not in client.TYPES:
+        return_error(
+            f"{SOURCE_NAME}: Got indicator_type {indicator_type} but expected "
+            f"one of {client.TYPES}"
+        )
+    limit = int(args.get("limit"))
+    if limit < 1:
+        limit = 1
+    if indicator_type == client.ALL_TYPE:
+        inside_limit = int(((limit + 1) / 2))
+        indicators_list = (
+            client.get_indicators_ip()[:inside_limit]
+            + client.get_indicators_domain()[:inside_limit]
+        )
```

Comment:
Your `_build_iterator()` method allows you to have both types chosen. Instead of making your life complicated here just do that. Sure, it won't guarantee that you'll get both types if you limit it to a small number but who cares.